src/resolvers/exotics/hosted-git-resolver.js

```diff
@@ -208,24 +116,15 @@ export default class HostedGitResolver extends ExoticResolver {
     // If we can access the files over HTTP then we should as it's MUCH faster than git
     // archive and tarball unarchiving. The HTTP API is only available for public repos
     // though.
-    if (await this.hasHTTPCapability(httpBaseUrl)) {
-      return this.resolveOverHTTP(httpUrl);
-    }
 
     // If the url is accessible over git archive then we should immediately delegate to
     // the git resolver.
     //
     // NOTE: Here we use a different url than when we delegate to the git resolver later on.
     // This is because `git archive` requires access over ssh and github only allows that
     // if you have write permissions
-    const sshGitUrl = Git.npmUrlToGitUrl(sshUrl);
-    if (await Git.hasArchiveCapability(sshGitUrl)) {
```

Comment:
Note I removed the Git check archive and new Git(...).
GitResolver creates `client = new Git(...)`, and `client.init()` does the same thing (check archive and resolve committish).
This just remove duplicated effort.